src/tiled/mapdocumentactionhandler.cpp

```diff
@@ -299,16 +302,64 @@ void MapDocumentActionHandler::selectAll()
     }
 }
 
+void MapDocumentActionHandler::selectInverse()
+{
+    if (!mMapDocument)
+        return;
+
+    Layer *layer = mMapDocument->currentLayer();
+    if (!layer)
+        return;
+
+    if (TileLayer *tileLayer = layer->asTileLayer()) {
+        QRegion all(tileLayer->x(), tileLayer->y(),
+                  tileLayer->width(), tileLayer->height());
+
+        /*if (mMapDocument->selectedArea().isEmpty()) {
+            selectAll();
+        } else if(mMapDocument->selectedArea() == all) {
+            selectNone();
+        } else {*/
```

Comment:
These special cases need not to be made, because that's what should be happening automatically, right? Please drop the commented out code.